### PR TITLE
add typescript types

### PR DESCRIPTION
**Pull request by jbarrus**
By submitting a PR to this repository, you agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md). Please see the [contributing guidelines](https://github.com/auth0/.github/blob/master/CONTRIBUTING.md) for how to create and submit a high-quality PR for this repo.

### Description

Adds typescript definitions to allow easier use with typescript projects.

### References

https://www.typescriptlang.org/docs/handbook/declaration-files/introduction.html

### Testing

Testing can be done by importing into a typescript project.

- [ ] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [X] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [X] The correct base branch is being used, if not `master`

### CODE REVIEW

**Comment by joshcanhelp** [comment]
@jbarrus - Thank you so much for this! We'll take a look this week and get back to you with feedback or an approval.

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import { AuthorizationParameters, TokenSet } from 'openid-client';
+import { Request, RequestHandler } from 'express';
+
+interface AuthParams {
+    issuerBaseURL?: string;
+    baseURL?: string;
+    clientID?: string;
+    clientSecret?: string;
+    clockTolerance?: number;
+    getUser?: (tokenSet: TokenSet) => any;
```
Comment (joshcanhelp) [inline index.d.ts]: Should this really be `any`? This function should return `undefined` or an `object` representing the user. Can we be more explicit here?

Comment (jbarrus) [reply index.d.ts]: Does it have to be an object or can it be a string or something else?

Comment (joshcanhelp) [reply index.d.ts]: Here's the default `getUser` implementation:

https://github.com/auth0/express-openid-connect/blob/master/lib/getUser.js

We should definitely be strict on what it gets back. Following this up to the parent library, by default, that returns `IdTokenClaims`, which extends `UserinfoResponse`. I would probably use the latter for now.

**Pull request merged**